lib/Protocol/Matrix.pm

```diff
@@ -106,25 +104,52 @@ character string. This is re-exported from L<MIME::Base64> for convenience.
 
    sign_json( $data, secret_key => $key, origin => $name, key_id => $id )
 
+or:
+
+   sign_json( $data,
+      eddsa_secret_key => $secret_key,
+      eddsa_public_key => $public_key,
+      origin => $name,
+      key_id => $id,
+   )
+
 Modifies the given HASH reference in-place to add a signature. This signature
 is created from the given key, and annotated as being from the given origin
 name and key ID. Existing signatures already in the hash are not disturbed.
 
-The C<$key> should be a plain byte string or L<Data::Locker> object obtained
-from L<Crypt::NaCl::Sodium::sign>'s C<keypair> method.
+The key can be specified in one of two ways: either with C<secret_key>, in
+which case the C<$key> should be a plain byte string or L<Data::Locker> object
+obtained from L<Crypt::NaCl::Sodium::sign>'s C<keypair> method; or with
+C<eddsa_secret_key> with a key returned by
+L<Crypt::Ed25519::eddsa_secret_key>. In the latter case C<eddsa_public_key> is
+optional, and the public key will be derived if not given.
```

Comment:
I found this a little hard to follow (with `secret_key`, `$secret_key`, `$key`, etc.) but after reading a few times it made sense.

Reply:
fair point. I can try to clarify this.